interpreter/src/functions.rs

```diff
@@ -1,14 +1,69 @@
 use crate::context::Context;
 use crate::magic::{Arguments, Identifier, This};
-use crate::objects::{Value, ValueType};
+use crate::objects::{CelType, ValueType};
 use crate::resolvers::{Argument, Resolver};
-use crate::ExecutionError;
+use crate::{ExecutionError, Value};
 use cel_parser::Expression;
 use std::cmp::Ordering;
 use std::convert::TryInto;
+use std::result::Result as StdResult;
 use std::sync::Arc;
 
-type Result<T> = std::result::Result<T, ExecutionError>;
+type Result<T> = StdResult<T, ExecutionError>;
+
+/// Returns the CEL type of the argument as a first-class value.
+///
+/// This function implements the standard CEL `type()` function, which returns
+/// the type of its argument in accordance with the CEL specification:
+///
+/// - When called on a type itself (like `type(int)` or `type(string)`), this returns the
+//    "type" type.
+/// - For other values, returns the type name as a string (e.g., `type(1)` returns "int")
+///
+/// This implementation also handles special cases for proper evaluation of expressions
+/// like `type(type(1)) == type(string)`.
+pub fn type_fn(_ftx: &FunctionContext, This(this): This<Value>) -> Result<Value> {
+    // For the value itself, we have two distinct cases:
+    // 1. If the value is a Type, we return the type of a type (which is "type" as a Type value)
+    // 2. If the value is a string, we need to check if it represents a type name (e.g., `type(string)`)
+    // 3. For all other values, we return their type name as a String value
+    match this {
+        // When type() is called on a type (like type(int)), it should return "type" as a Type value
+        Value::Type(_) => Ok(Value::Type(CelType::Type)),
+        // For strings, we need to check if they represent a type name
+        Value::String(s) => {
+            // If the string value is a type name (like "string", "int", etc.),
+            // we should return the Type type for consistency with type(string), type(int), etc.
+            if CelType::try_from_string(s.as_str()).is_some() {
+                Ok(Value::Type(CelType::Type))
+            } else {
+                Ok(Value::String("string".to_string().into()))
+            }
+        }
+        Value::Null => Ok(Value::String("null".to_string().into())),
+        Value::Bool(_) => Ok(Value::String("bool".to_string().into())),
+        Value::Int(_) => Ok(Value::String("int".to_string().into())),
+        Value::UInt(_) => Ok(Value::String("uint".to_string().into())),
+        Value::Float(_) => Ok(Value::String("float".to_string().into())),
+        Value::Bytes(_) => Ok(Value::String("bytes".to_string().into())),
```

Comment:
1. It looks like `type(int)` would return `CelType::Int`, but `type(10)` would return `int`. Shouldn't they return the same?
2. Shouldn't `type("foobar")` return `CelType::String`?

Reply:
1. Does that mean CEL Playground would resolve `type(int) == type(10)` as true? (It gives me false).
2. Is there an easy place to test that?